lib/puppet/provider/cisco_fabricpath_topology/nxapi.rb

```diff
@@ -0,0 +1,143 @@
+# The NXAPI provider for cisco fabricapth topology.
+#
+# January, 2016
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
+rescue LoadError # seen on master, not on agent
+  # See longstanding Puppet issues #4248, #7316, #14073, #14149, etc. Ugh.
+  require File.expand_path(File.join(File.dirname(__FILE__), '..', '..', '..',
+                                     'puppet_x', 'cisco', 'autogen.rb'))
+end
+
+Puppet::Type.type(:cisco_fabricpath_topology).provide(:nxapi) do
+  desc 'The new NXAPI provider.'
+
+  confine feature: :cisco_node_utils
+  defaultfor operatingsystem: :nexus
+
+  mk_resource_methods
+
+  TOPO_NON_BOOL_PROPS = [:member_vlans, :topo_name]
+  TOPO_ALL_PROPS = TOPO_NON_BOOL_PROPS
+
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:non_bool, self, '@topo',
+                                            TOPO_NON_BOOL_PROPS)
+
+  def self.properties_get(topo_id, t)
+    debug "Checking instance, topo #{topo_id}"
+    current_state = {
+      topo_id:      topo_id,
+      name:         topo_id,
+      member_vlans: [],
```

Comment:
What's the reason for setting this to `[]` here when you're also setting it on line 53?